workflow/common/util.go

```diff
@@ -183,6 +186,26 @@ func ProcessArgs(tmpl *wfv1.Template, args wfv1.ArgumentsProvider, globalParams,
 	return SubstituteParams(newTmpl, globalParams, localParams)
 }
 
+// substituteConfigMapKeyRefParams check if ConfigMapKeyRef's Name and Value are params and perform the substitution
+func substituteConfigMapKeyRefParams(inParam wfv1.Parameter, globalParams Parameters) (string, string) {
+	name := inParam.ValueFrom.ConfigMapKeyRef.Name
+	key := inParam.ValueFrom.ConfigMapKeyRef.Key
+
+	if strings.HasPrefix(name, "{{") && strings.HasSuffix(name, "}}") {
+		k := strings.TrimSuffix(strings.TrimPrefix(inParam.ValueFrom.ConfigMapKeyRef.Name, "{{"), "}}")
+		if v, ok := globalParams[k]; ok {
+			name = v
+		}
+	}
+	if strings.HasPrefix(key, "{{") && strings.HasPrefix(key, "}}") {
+		k := strings.TrimSuffix(strings.TrimPrefix(inParam.ValueFrom.ConfigMapKeyRef.Key, "{{"), "}}")
+		if v, ok := globalParams[k]; ok {
+			key = v
+		}
+	}
```

Comment:
Redundant?

Reply:
I refactored a bit the code to reduce redundancy.